### PR TITLE
[IOP] merge changes from IOP14

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

**Review by EskoDijk — APPROVED** [review]
Changes look good, I only have a few extra comment lines to commit to IOP14. (Can be done before merge - I still need to get access to committing - or else after the merge.)

File: src/library/cwt.hpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ static const int kGrantType           = 33;
 static const int kGrantTypeClientCred = 2;
 static const int kClientId            = 24;
 static const int kReqAud              = 3;
-static const int kReqCnf              = 12;
+static const int kReqCnf              = 4;
```
[the remark this answers is not in the document]
Comment (jwhui) [reply src/library/cwt.hpp:54]: @EskoDijk, thanks for the review! I'll go ahead and merge this now - please submit a separate PR with your changes.

File: src/library/token_manager.hpp [hunk 2]
```diff
@@ -43,45 +43,108 @@ namespace ot {
...
+ */
 class TokenManager
 {
 public:
     explicit TokenManager(struct event_base *aEventBase);
     ~TokenManager();
 
-    // Initialized with Commissioner configuration.
+    /**
+     * This method initialize the Token Manager.
```
Comment (jwhui) [inline src/library/token_manager.hpp:57]:
```suggestion
     * This method initializes the Token Manager.
```

File: src/library/token_manager.hpp [hunk 3]
```diff
@@ -43,45 +43,108 @@ namespace ot {
...
-    bool IsValid() const { return mToken.IsValid(); }
-
-    // Set the signed Commissioner Token when the signature verification succeed.
-    // @param[in] aSignedToken  A COSE-signed Commissioner Token.
-    // @param[in] aCert         A certificate associates to the signing key of @p aSignedToken.
-    Error SetToken(const ByteArray &aSignedToken, const ByteArray &aCert);
+    /**
+     * This method sets the Commissioner Token (COM_TOK).
+     *
+     * This method will firstly validate the Commissioner Token against the Domain CA
```
Comment (jwhui) [inline src/library/token_manager.hpp:67]:
```suggestion
     * This method will first validate the Commissioner Token against the Domain CA
```

File: src/library/token_manager.hpp [hunk 4]
```diff
@@ -43,45 +43,108 @@ namespace ot {
...
+     *
+     */
     static Error ParsePublicKey(mbedtls_pk_context &aPublicKey, const ByteArray &aCert);
 
-    // Parse the private key from the PEM/DER encoded private key.
+    /**
+     * This method parse private key from PEM/DER encoded certificate.
+     *
+     * @param[out]  aPrivateKey  The parsed private key.
+     * @param[in]   aCert        The given PEM/DER encoded certificate.
```
Comment (jwhui) [inline src/library/token_manager.hpp:145]: Wrong param name.

File: src/library/token_manager.hpp [hunk 5]
```diff
@@ -43,45 +43,108 @@ namespace ot {
...
+     * is enabled, the token will be always be accepted if `aAlwaysAccept` is true.
+     * Otherwise, the Commissioner Token validation must succeed before accepting it.
+     *
+     * @param[in]  aSignedToken   The COSE-signed Commissioner Token to set to.
+     * @param[in]  aAlwaysAccept  A boolean indicates whether we always accept the given
+     *                            Commissioner Token. This paramater has effect only
+     *                            when `OT_COMM_CONFIG_REFERENCE_DEVICE_ENABLE` is enabled.
+     *
+     */
+    Error SetToken(const ByteArray &aSignedToken, bool aAlwaysAccept = false);
```
Comment (jwhui) [inline src/library/token_manager.hpp:78]: Document return values here and below?

**Review by jwhui — APPROVED** [review]

**Pull request merged**